packages/model-serving/src/components/deploymentWizard/fields/ModelLocationSelectField.tsx

```diff
@@ -0,0 +1,158 @@
+import { z, type ZodIssue } from 'zod';
+import SimpleSelect from '@odh-dashboard/internal/components/SimpleSelect';
+import { FieldValidationProps } from '@odh-dashboard/internal/hooks/useZodFormValidation';
+import { ZodErrorHelperText } from '@odh-dashboard/internal/components/ZodErrorFormHelperText';
+import {
+  Connection,
+  ConnectionTypeConfigMapObj,
+} from '@odh-dashboard/internal/concepts/connectionTypes/types';
+import { getConnectionTypeRef } from '@odh-dashboard/internal/concepts/connectionTypes/utils';
+import { getResourceNameFromK8sResource } from '@odh-dashboard/internal/concepts/k8s/utils';
+import { useWatchConnectionTypes } from '@odh-dashboard/internal/utilities/useWatchConnectionTypes';
+import { LabeledConnection } from '@odh-dashboard/internal/pages/modelServing/screens/types';
+import { KnownLabels } from '@odh-dashboard/internal/k8sTypes';
+import { ModelLocationInputFields } from './ModelLocationInputFields';
+import { ModelLocationData, ModelLocationType } from './modelLocationFields/types';
+
+// Schema
+export const modelLocationSelectFieldSchema = z.enum(
+  [ModelLocationType.EXISTING, ModelLocationType.NEW, ModelLocationType.PVC],
+  {
+    // eslint-disable-next-line @typescript-eslint/naming-convention, camelcase
+    required_error: 'Select a model location.',
+  },
+);
+
+export type ModelLocationFieldData = z.infer<typeof modelLocationSelectFieldSchema>;
+export const isValidModelLocation = (value: string): value is ModelLocationFieldData =>
+  value === ModelLocationType.EXISTING ||
+  value === ModelLocationType.NEW ||
+  value === ModelLocationType.PVC;
+
+// Hooks
+export type ModelLocationField = {
+  data: ModelLocationFieldData | undefined;
+  setData: (data: ModelLocationFieldData) => void;
+  connections: LabeledConnection[];
+  setSelectedConnection: (
+    connection: LabeledConnection | undefined,
+    connectionTypes: ConnectionTypeConfigMapObj[],
+  ) => void;
+  selectedConnection: LabeledConnection | undefined;
+};
+// Component
+
+type ModelLocationSelectFieldProps = {
+  modelLocation?: ModelLocationData['type'];
+  validationProps?: FieldValidationProps;
+  validationIssues?: ZodIssue[];
+  connections: LabeledConnection[];
+  setModelLocationData: (data: ModelLocationData | undefined) => void;
+  resetModelLocationData: () => void;
+  modelLocationData?: ModelLocationData;
+  initSelectedConnection: LabeledConnection | undefined;
+};
+export const ModelLocationSelectField: React.FC<ModelLocationSelectFieldProps> = ({
+  modelLocation,
+  validationProps,
+  validationIssues = [],
+  connections,
+  setModelLocationData,
+  resetModelLocationData,
+  modelLocationData,
+  initSelectedConnection,
+}) => {
+  const [selectedConnectionState, setSelectedConnection] = React.useState<Connection | undefined>(
+    initSelectedConnection?.connection ?? undefined,
+  );
+  const [modelServingConnectionTypes] = useWatchConnectionTypes(true);
```

Comment:
I think ideally we would load connectionTypes outside of the field and pass them to the wizard so we can front load dependencies before getting to each step. We can change this in a future PR though